src/reactive.coffee

```diff
@@ -343,13 +343,62 @@ rxFactory = (_, $) ->
     splice: (index, count, additions...) -> @spliceArray(index, count, additions)
     insert: (x, index) -> @splice(index, 0, x)
     remove: (x) ->
-      i = _(@raw()).indexOf(x)
+      i = _(rx.snap => @all()).indexOf(x)
```

Comment:
Why move to all()? Slower, and the original use of `raw()` was safe / non-mutating

Same various points below

Also really need to break these out into separate commits